app/uk/gov/hmrc/customs/notification/services/UnblockPollerService.scala

```diff
@@ -67,40 +67,47 @@ class UnblockPollerService @Inject()(config: CustomsNotificationConfig,
     }
   }
 
-  private def pushOrPull(workItem: WorkItem[NotificationWorkItem]): Future[Boolean] = {
+  private def pushOrPull(workItem: WorkItem[NotificationWorkItem]): Future[Response] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrier()
 
-    pushOrPullService.send(workItem.item).map[Boolean] {
+    pushOrPullService.send(workItem.item).flatMap {
       case Right(connector) =>
         notificationWorkItemRepo.setCompletedStatus(workItem.id, Succeeded)
         logger.info(s"Unblock pilot for $connector succeeded. CsId = ${workItem.item.clientSubscriptionId.toString}. Setting work item status ${Succeeded.name} for $workItem")
-        true
+        Future.successful(Success)
       case Left(PushOrPullError(connector, resultError)) =>
         logger.info(s"Unblock pilot for $connector failed with error $resultError. CsId = ${workItem.item.clientSubscriptionId.toString}. Setting work item status back to ${PermanentlyFailed.name} for $workItem")
         (for {
           _ <- notificationWorkItemRepo.incrementFailureCount(workItem.id)
-          _ <- {
+          status <- {
             resultError match {
-              case httpResultError: HttpResultError if httpResultError.is3xx || httpResultError.is4xx =>
+              case error@HttpResultError(status, _) if error.is3xx || error.is4xx =>
                 val availableAt = dateTimeService.zonedDateTimeUtc.plusMinutes(customsNotificationConfig.notificationConfig.nonBlockingRetryAfterMinutes)
-                logger.error(s"Status response ${httpResultError.status} received while trying unblock pilot, setting availableAt to $availableAt")
-                notificationWorkItemRepo.setCompletedStatusWithAvailableAt(workItem.id, PermanentlyFailed, httpResultError.status, availableAt)
+                logger.error(s"Status response $status received while trying unblock pilot, setting availableAt to $availableAt and status to Failed")
+                notificationWorkItemRepo.setCompletedStatusWithAvailableAt(workItem.id, Failed, status, availableAt)
+                  .map(_ => ClientError)
               case HttpResultError(status, _) =>
                 notificationWorkItemRepo.setPermanentlyFailed(workItem.id, status)
+                .map(_ =>ServerError)
               case NonHttpError(cause) =>
                 logger.error(s"Error received while unblocking notification: ${cause.getMessage}")
-                Future.successful(())
+                Future.successful(ServerError)
             }
```

Comment:
It's a suggestion. It would be nice to have the whole resultError matching in a method. So the code looks tide. There are so many closing curly braces. Very hard to see which black is doing what.